### PR TITLE
[Redirect] Allows for Class and Symbol Controller name

### PR DESCRIPTION
**Pull request by eliasjpr**
Issue: https://github.com/amberframework/amber/issues/353

> As a developer I would like to pass and full controller name to a
redirect as a class or symbol and allow for the redirection to happen

Example

`redirect_to controller: :HomeController, action: :index`

- Would be nice to accept HelloController::Index or
Admin::UserController:Index

### CODE REVIEW

**Comment by eliasjpr** [comment]
That would be a great idea. We should also have the ‘link_to’ jasper helpers user class name

**Comment by elorest** [comment]
We need to create url_for and have link_to and redirect_to translate links with that like in rails.

**Comment by eliasjpr** [comment]
A couple of changes introduced in the last 2 commits. I have decided to simplified the redirect api. 

There is only 3 possible ways to redirect now:
- Redirecting to full URL will be redirect_to "http://google.com", params. String based redirect are assumed that the user is passing the full path or url to redirect.
- Redirecting controller action `redirect_to PostController, :index`, `redirect_to :post, :index`
- Redirecting withing same controller `redirect_to :index` whichever controller this is called from it will be implicitly added.

Why?

- This keeps things simple and coherent and simple and short.
- Supporting `redirect_to :PostController, :index` or `redirect_to "PostController", :index` makes things messy and confusing
- People can pass controller name as symbols without the "controller" part, keeps it short.

File: src/amber/router/route.cr [hunk 1]
```diff
@@ -49,7 +49,11 @@ module Amber
       {result, params}
     end
 
-    def match?(controller, action)
+    def match?(controller : Class, action : Symbol)
```
Comment (eliasjpr) [inline src/amber/router/route.cr]: This is separated into 2 method so we don't have to perform checks or substitutions for `/Controller$/`

**Pull request merged**